plugins/filter/filters.py

```diff
@@ -228,23 +228,38 @@ def listener_properties(self, listeners_dict, default_ssl_enabled,
                 final_dict['listener.name.' + listener_name + '.sasl.oauthbearer.sub.claim.name'] = oauth_sub_claim
 
             if self.normalize_sasl_protocol(listeners_dict[listener].get('sasl_protocol', default_sasl_protocol)) == 'OAUTHBEARER' \
                     and oauth_enabled and not rbac_enabled:
-                final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
-                    'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required unsecuredLoginStringClaim_sub="thePrincipalName";'
                 final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.server.callback.handler.class'] =\
                     'org.apache.kafka.common.security.oauthbearer.secured.OAuthBearerValidatorCallbackHandler'
                 final_dict['listener.name.' + listener_name + '.sasl.oauthbearer.jwks.endpoint.url'] = oauth_jwks_uri
                 final_dict['listener.name.' + listener_name + '.sasl.oauthbearer.sub.claim.name'] = oauth_sub_claim
 
+            if self.normalize_sasl_protocol(listeners_dict[listener].get('sasl_protocol', default_sasl_protocol)) == 'OAUTHBEARER' and \
+                    oauth_enabled:
+                if rbac_enabled and idp_self_signed:
+                    final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
+                        'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required publicKeyPath=\"' + \
+                        oauth_pem_path + '\" ssl.truststore.location=\"' + kafka_broker_truststore_path + \
+                        '\" ssl.truststore.password=\"' + kafka_broker_truststore_storepass + '\" unsecuredLoginStringClaim_sub="thePrincipalName";'
+                if (rbac_enabled and (not idp_self_signed)):
+                    final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
+                        'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required publicKeyPath=\"' + \
+                        oauth_pem_path + '\" unsecuredLoginStringClaim_sub="thePrincipalName";'
+                if ((not rbac_enabled) and idp_self_signed):
+                    final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
+                        'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required' + \
+                        ' ssl.truststore.location=\"' + kafka_broker_truststore_path + \
+                        '\" ssl.truststore.password=\"' + kafka_broker_truststore_storepass + '\" unsecuredLoginStringClaim_sub="thePrincipalName";'
+                if ((not rbac_enabled) and (not idp_self_signed)):
+                    final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
+                        'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required unsecuredLoginStringClaim_sub="thePrincipalName";'
+
```

Comment:
in all 4 conditions (i.e rbac_enabled x idp_self_signed (1,1), (1,0), (0,1), (0,0) ) 2 things are common `OAuthBearerLoginModule & unsecuredLoginStringClaim_sub` So that can be taken out. 
And also instead of using 4 such conditions probably it would be simpler if use something like
if rbac_enabled then append the configs that are coming from rbac.
if idp_self_signe then append the configs that are coming from it.

Reply:
I'm not sure if append works in our filters.py file, if it does work then it's best to revamp the entire file but used it this way to maintain uniformity for now

Reply:
I am confused here bcos what I was saying would need string concat. So basically I meant something like ->
```                
final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] = 'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule'          
if rbac_enabled:
    final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] += 'required publicKeyPath=\"' + oauth_cert_path 
if idp_self_signed:
    final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] += ' ssl.truststore.location=\"' + kafka_broker_truststore_path + \
        '\" ssl.truststore.password=\"' + kafka_broker_truststore_storepass 
final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] += 'unsecuredLoginStringClaim_sub="thePrincipalName";'
```

Reply:
I see, yes this might work but will again need some rounds of testing, I don't think it adds much value for now considering it's not as elegant either. Do you think it makes that much difference?

Reply:
imv it makes the code more clear that what configs are getting added due to what conditions. Down the line it would be easier to maintain it ig